CommandRewriteDesignDoc.md

```diff
@@ -0,0 +1,293 @@
+# WPILib Command Framework Rewrite
+Design document
+
+The WPILib Command-based framework has been a staple of FRC Java and C++ robot code development for years, offering teams a relatively clean and sensible API for factoring complex robot code into simple, encapsulated chunks that interact in clear, comprehensible ways.  In addition to helping teams avoid the often substantial overhead of rolling their own state machine logic, it also serves as an important pedagogical tool for teaching teams about encapsulation, control flow, and design patterns.
+
+However, the libraries used in the framework are now more than a decade old.  Many of the particular choices in their implementation are obsolete, and others are messy, difficult-to-maintain, and/or confusingly-conceived.  Given the confusing nature of much of the implementation, bringing the API up to modern standards is a difficult task, and the libraries have ended up in a "stasis" of sorts.
+
+To document will describe the goals, design choices, and implementation of a ground-up rewrite of the Command-based libraries aimed at alleviating these issues.
+
+
+## Why rewrite the framework? (Problem Summary)
```

Comment:
Can you please rewrite this section to be less condemning of the old framework? Teams have successfully used it for 10+ years and I think writing about it in this way is not good for it or WPILib. See some of my inline comments.